src/Media.Plugin/UWP/NewMediaImplementation.cs

```diff
@@ -200,8 +200,11 @@ async Task SaveToAlbum(StorageFile file, StorageFolder folder, string directory
         {
             try
             {
-                if (IsValidPathName(directory))
-                    folder = await folder.CreateFolderAsync(directory, CreationCollisionOption.GenerateUniqueName);
+                if (!IsValidFileName(directory))
+                {
+                    directory = string.Join("", directory.Split(Path.GetInvalidFileNameChars()));
+                }
+                folder = await folder.CreateFolderAsync(directory, CreationCollisionOption.GenerateUniqueName);
```

Comment:
`directory` could be `null` or `string.Empty` at this point

Reply:
Oh true. And if so, should I just put the image to the Saved Pictures folder without a subfolder?

Reply:
exactly